DigitalLearningSolutions.Web/Controllers/TrackingSystem/CourseSetup/AdminFieldsController.cs

```diff
@@ -146,6 +147,97 @@ BulkAdminFieldAnswersViewModel model
             );
         }
 
+        [HttpGet]
+        [Route("{customisationId:int}/AdminFields/Add/New")]
+        public IActionResult AddAdminFieldNew(int customisationId)
+        {
+            TempData.Clear();
+
+            var model = new AddAdminFieldViewModel(customisationId);
+
+            SetAddAdminFieldTempData(model);
+
+            return RedirectToAction("AddAdminField", new { customisationId = model.CustomisationId });
+        }
+
+        [HttpGet]
+        [Route("{customisationId:int}/AdminFields/Add")]
+        public IActionResult AddAdminField(int customisationId)
+        {
+            var addAdminFieldData = TempData.Peek<AddAdminFieldData>()!;
+
+            SetViewModelCoursePromptNameOptions(addAdminFieldData.AddModel);
+
+            var model = addAdminFieldData?.AddModel ?? new AddAdminFieldViewModel(customisationId);
+
+            return View(model);
+        }
+
+        [HttpPost]
+        [Route("{customisationId}/AdminFields/Add")]
+        public IActionResult AddAdminField(int customisationId, AddAdminFieldViewModel model, string action)
+        {
+            if (!ModelState.IsValid)

```

Comment:
What if the user wants to delete/add an answer but they haven't selected an admin field yet?

Reply:
I've made the Answer field required in the View model but cleared the Answer field error in appropriate places so the user can add or remove an answer or navigate to the bulk page without first selecting a field name